From 610dd2eef091fb17377663c69f96d1d155da523e Mon Sep 17 00:00:00 2001
From: Pan Junzhong <junzhong.pan@spacemit.com>
Date: Wed, 8 May 2024 20:20:57 +0800
Subject: [PATCH 0639/1206] USB: xhci-plat: fix legacy PHY double init

Divers such as dwc3 which manages PHYs themself as the PHYs would
be initialised twice, something which specifically can lead to
resources being left enabled during suspend (e.g. with the
usb_phy_generic PHY driver).

As the dwc3 driver uses driver-name matching for the xhci platform
device, fix this by only looking up and initialising PHYs for devices
that have been matched using OF.

This commit is from upstream, picking codes for linux 6.1, link:
https://lore.kernel.org/r/20231103164323.14294-1-johan+linaro@kernel.org
Author: Johan Hovold <johan+linaro@kernel.org>

Change-Id: Id4472175fb72b3da2cb4ee108ec1afe6a4e87da0
---
 drivers/usb/host/xhci-plat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index db109b570c5c..552343f4a9ee 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -270,7 +270,7 @@ int xhci_plat_probe(struct platform_device *pdev, struct device *sysdev, const s
 	}
 
 	/*
-	 * Drivers such as dwc3 manages PHYs themself (and rely on driver name
+	 * Drivers such as dwc3 manages PHYs themself (and rely on drivers name
 	 * matching for the xhci platform device).
 	 */
 	of_match = of_match_device(pdev->dev.driver->of_match_table, &pdev->dev);
-- 
2.47.0

